### PR TITLE
Fix flaky tests (stateless, integration)

### PR DESCRIPTION
**Pull request by fm4v**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

Changes:
1. Set the host user for MySQL Docker containers. When running pytest locally, tests fail because of Permission denied: 'logs'.
2. Rich exception message raised from `ClickHouseInstance.query_with_retry()`
3. (attempt to) Fix some flaky integration and stateless tests

Failed reports is [here](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSBjaGVja19uYW1lIExJS0UgJ0ludGVncmF0aW9uJScKICAgIEFORCBjaGVja19zdGFydF90aW1lID49IG5vdygpIC0gSU5URVJWQUwgNjAgSE9VUgogICAgQU5EIHB1bGxfcmVxdWVzdF9udW1iZXIgPSAwCiAgICBBTkQgdGVzdF9zdGF0dXMgIT0gJ1NLSVBQRUQnCiAgICBBTkQgdGVzdF9zdGF0dXMgTElLRSAnRiUnCiAgICBBTkQgY2hlY2tfc3RhdHVzICE9ICdzdWNjZXNzJwpPUkRFUiBCWSBjaGVja19uYW1lLCB0ZXN0X25hbWUsIGNoZWNrX3N0YXJ0X3RpbWU=)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 3939a6ae5bcb261fd849367bde8717f4492b905d with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61816/3939a6ae5bcb261fd849367bde8717f4492b905d/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1">❌ failure</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/02115_rewrite_local_join_right_distribute_table.sql [hunk 1]
```diff
@@ -23,6 +23,10 @@ select t1.* from t1_all t1 join t2_all t2 on t1.a = t2.a ORDER BY t1.a;
 
 SELECT '-';
 
+-- make sure data is fully written when reading from distributed
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/02115_rewrite_local_join_right_distribute_table.sql:26]: But the data is fully written to a MergeTree table, regardless of OPTIMIZE.

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/02115_rewrite_local_join_right_distribute_table.sql (hunk 1 above)
Comment (fm4v) [reply tests/queries/0_stateless/02115_rewrite_local_join_right_distribute_table.sql:26]: I'm not sure about my fix, but I want to keep an eye on it's flakiness.

File: tests/integration/test_ttl_move/test.py [hunk 2]
```diff
@@ -1635,7 +1635,7 @@ def optimize_table(num):
         optimize_table(20)
 
         if positive:
-            assert check_used_disks_with_retry(node1, name, set(["external"]), 50)
+            assert check_used_disks_with_retry(node1, name, set(["external"]), 100)
```
Comment (azat) [inline tests/integration/test_ttl_move/test.py:1638]: Does increasing number of retries fixes something? I doubt

Comment (azat) [reply tests/integration/test_ttl_move/test.py:1638]: Anyway it still fails - https://s3.amazonaws.com/clickhouse-test-reports/66671/2f00c962711e13ca00af324366421fe4593b4ce6/integration_tests__tsan__[5_6].html

Comment (azat) [reply tests/integration/test_ttl_move/test.py:1638]: I don't even understand why retries are needed after OPTIMIZE FINAL

Comment (azat) [reply tests/integration/test_ttl_move/test.py:1638]: https://github.com/ClickHouse/ClickHouse/pull/67747

**Pull request merged**